crowbar_framework/app/models/ceph_service.rb

```diff
@@ -137,9 +137,9 @@ def create_proposal
     end
     mon_nodes = mon_nodes.take(mon_nodes.length > 2 ? 3 : 1)
 
-    mds_node = select_nodes_for_role(nodes, "ceph-mds", "controller").first
+    mds_node = select_nodes_for_role(nodes, "ceph-mds", "storage").first
 
-    radosgw_node = select_nodes_for_role(nodes, "ceph-radosgw", "controller").first
+    radosgw_node = select_nodes_for_role(nodes, "ceph-radosgw", "storage").first
```

Comment:
@jsuchome I believe we documented somewhere that mixing openstack roles with ceph roles is unsupported, so avoiding that via constraints checks should be good for now (it might break existing customers violating that though).